features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/RuntimeConfiguration.jsx

```diff
@@ -126,41 +126,74 @@ const useStyles = makeStyles((theme) => ({
+        apiConfigJson = {
+            id: api.id,
+            name: api.name,
+            description: api.description,
+            lifeCycleStatus: api.lifeCycleStatus,
+            accessControl: api.accessControl,
+            authorizationHeader: api.authorizationHeader,
+            responseCachingEnabled: api.responseCachingEnabled,
+            cacheTimeout: api.cacheTimeout,
+            visibility: api.visibility,
+            isDefaultVersion: api.isDefaultVersion,
+            enableSchemaValidation: api.enableSchemaValidation,
+            accessControlRoles: [...api.accessControlRoles],
+            visibleRoles: [...api.visibleRoles],
+            tags: [...api.tags],
+            maxTps: api.maxTps,
+            wsdlUrl: api.wsdlUrl,
+            transport: [...api.transport],
+            securityScheme: [...api.securityScheme],
+            keyManagers: [...(api.keyManagers || [])],
+            corsConfiguration: {
+                corsConfigurationEnabled: api.corsConfiguration.corsConfigurationEnabled,
+                accessControlAllowCredentials: api.corsConfiguration.accessControlAllowCredentials,
+                accessControlAllowOrigins: [...api.corsConfiguration.accessControlAllowOrigins],
+                accessControlAllowHeaders: [...api.corsConfiguration.accessControlAllowHeaders],
+                accessControlAllowMethods: [...api.corsConfiguration.accessControlAllowMethods],
+            },
+        };
+    } else {
+        apiConfigJson = {
+            id: api.id,
+            name: api.name,
+            description: api.description,
+            lifeCycleStatus: api.lifeCycleStatus,
+            accessControl: api.accessControl,
+            authorizationHeader: api.authorizationHeader,
+            responseCachingEnabled: api.responseCachingEnabled,
+            cacheTimeout: api.cacheTimeout,
+            visibility: api.visibility,
+            isDefaultVersion: api.isDefaultVersion,
+            enableSchemaValidation: api.enableSchemaValidation,
+            accessControlRoles: [...api.accessControlRoles],
+            visibleRoles: [...api.visibleRoles],
+            tags: [...api.tags],
+            maxTps: api.maxTps,
+            wsdlUrl: api.wsdlUrl,
+            transport: [...api.transport],
+            securityScheme: [...api.securityScheme],
+            keyManagers: [...(api.keyManagers || [])],
+            corsConfiguration: {
+                corsConfigurationEnabled: api.corsConfiguration.corsConfigurationEnabled,
+                accessControlAllowCredentials: api.corsConfiguration.accessControlAllowCredentials,
+                accessControlAllowOrigins: [...api.corsConfiguration.accessControlAllowOrigins],
+                accessControlAllowHeaders: [...api.corsConfiguration.accessControlAllowHeaders],
+                accessControlAllowMethods: [...api.corsConfiguration.accessControlAllowMethods],
+            },
+            apiwebsubSubscriptionConfiguration: {
```

Comment:
Can't we separate out the API product-related config params from the API like `apiwebsubSubscriptionConfiguration` and prevent repeating the same properties twice in the API and API product config JSON?

Reply:
fixed